### PR TITLE
Wip/hughsie/util cmd

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-tool.c [hunk 1]
```diff
@@ -1515,10 +1396,6 @@ main (int argc, char *argv[])
 	g_signal_connect (priv->cancellable, "cancelled",
 			  G_CALLBACK (fu_util_cancelled_cb), priv);
 
-	/* sort by command name */
-	g_ptr_array_sort (priv->cmd_array,
-			  (GCompareFunc) fu_sort_command_name_cb);
-
```
Comment (superm1) [inline src/fu-tool.c]: Why did you move the sorting into when the command was added?  It seems that will cause it to be called a lot more than is necessary.

Comment (hughsie) [reply src/fu-tool.c]: Good question. I made it automatic as the overhead is literally not measurable, and it's one less thing for the CLI to forget to do. If you think it's better explicit (e.g. a CLI might not *want* the commands sorted), I can move it out again.

File: src/fu-util.c [hunk 2]
```diff
@@ -302,54 +181,6 @@ fu_util_prompt_for_device (FuUtilPrivate *priv, GError **error)
...
-			return FALSE;
-		}
-		g_object_set (priv->soup_session, SOUP_SESSION_PROXY_URI, proxy_uri, NULL);
-	}
-
-	/* this disables the double-compression of the firmware.xml.gz file */
-	soup_session_remove_feature_by_type (priv->soup_session, SOUP_TYPE_CONTENT_DECODER);
-	return TRUE;
-}
-
```
Comment (superm1) [inline src/fu-util.c]: `fu-tool` intentionally forks out to `wget` or `curl` to be safer in terms of network comms.  I'm not sure there is a reason to move this into the common source file.

Comment (hughsie) [reply src/fu-util.c]: I'm about to use it for a new CLI -- I don't mind including this commit on that branch if you want. Just seemed nicer on this PR.

File: src/fu-tool.c (hunk 1 above)
Comment (superm1) [reply src/fu-tool.c]: Are you thinking we might have a 3rd tool some day and don't want to forget it there?  It seems like good coding practice to not call things that run every time the app is opened more than is needed even if it's not a measurable impact.

File: src/fu-util.c (hunk 2 above)
Comment (superm1) [reply src/fu-util.c]: Oh as long as there is bigger picture stuff for it, that's fine then.

File: src/fu-tool.c (hunk 1 above)
Comment (hughsie) [reply src/fu-tool.c]: Fair enough, I'll move it out and be explicit.

Comment (hughsie) [reply src/fu-tool.c]: In fact, after doing this; do we really want the commands to be sorted? If we leave them alone, we can group them in the sections of the things they're doing, e.g. `enable-remote` gets put before `disable-remote`.

Comment (superm1) [reply src/fu-tool.c]: That's a great point.  I do think they're better grouped on what they do.

Comment (hughsie) [reply src/fu-tool.c]: Okay, I've put the sorting back in, and in a different PR we can discuss the grouping, and disable the sorting.
